docs/examples/application_state/using_application_state.py

```diff
@@ -27,7 +27,7 @@ async def my_middleware(scope: "Scope", receive: "Receive", send: "Send") -> Non
     return my_middleware
 
 
-def my_dependency(state: State) -> Any:
+async def my_dependency(state: State) -> Any:
     """Dependencies can receive state via injection."""
     logger.info("state value in dependency: %s", state.value)
```

Comment:
Why convert it to async? Just wrap it in provide in the examples ), except one case that demoes the use without Provide.